### PR TITLE
Pr loading

### PR DESCRIPTION
**Pull request by janefwp**
add loader

### CODE REVIEW

File: src/app/shared/loader/loader.component.html [hunk 1]
```diff
@@ -0,0 +1,5 @@
+<div class="d-flex justify-content-center">
```
Comment (whywhy36) [inline src/app/shared/loader/loader.component.html:1]: 👍

File: src/app/shared/shared.component.html [hunk 2]
```diff
@@ -0,0 +1,3 @@
+<p>
+  shared works!
```
Comment (whywhy36) [inline src/app/shared/shared.component.html:2]: it's nice to see some shared components extracted.

File: src/app/todo/todo-list/todo-list.component.ts [hunk 3]
```diff
@@ -17,13 +16,15 @@ import {  addTodoReq,  delTodoReq, getTodosReq, updateTodoReq } from 'src/app/st
...
+  isLoading$: Observable<boolean>;
+
   constructor(
     private logger: NGXLogger,
     private store: Store
   ) { 
-    this.todos$=this.store.select(selectTodos)
+    this.todos$ = this.store.select(selectTodos)
     this.logger.info(this.todos$)
+    this.isLoading$ = this.store.select(selectLoading);
```
Comment (whywhy36) [inline src/app/todo/todo-list/todo-list.component.ts:29]: not a major thing. just a suggestion in the future.

we would like to move line 27 above line 26. And use `.debug` if you really want to print out the object.

File: src/app/user/profile/profile.component.ts [hunk 4]
```diff
@@ -15,11 +16,14 @@ import { User } from '../models/user';
...
   user$: Observable<User>;
+  isLoading$: Observable<boolean>;
   constructor(
     private userauthservice: UserauthService,
     private store: Store
   ) {
-    this.user$=this.store.select(selectUser);
+    this.user$ = this.store.select(selectUser);
+    this.isLoading$ = this.store.select(selectLoading)
+    console.log(this.isLoading$)
```
Comment (whywhy36) [inline src/app/user/profile/profile.component.ts]: you have a logger now :-)

File: src/app/todo/todo-form/todo-form.component.ts [hunk 5]
```diff
@@ -1,18 +1,46 @@
...
   selector: 'app-todo-form',
   templateUrl: './todo-form.component.html',
   styleUrls: ['./todo-form.component.css']
 })
 export class TodoFormComponent {
+  constructor(private modalService: NgbModal) {}
+
   @Output() newItemEvent = new EventEmitter<Todo>();
   todo = {title:'', description:'', deadline: new Date()}; 
+  IsmodelShow = false;
```
Comment (whywhy36) [inline src/app/todo/todo-form/todo-form.component.ts]: maybe use `deactivate guard`?

**Review by whywhy36 — APPROVED** [review]
good one. 

few minor comments - the only thing that is important is to use DeactivateGuard for the dirty checking exit.

File: src/app/todo/todo-form/todo-form.component.ts (hunk 5 above)
Comment (janefwp) [reply src/app/todo/todo-form/todo-form.component.ts]: I used another method to fix this issue and submitted to this branch

**Pull request merged**